### ISSUE TITLE
equation solver does not give solution in certain cases, blocking problem on exams

### ISSUE THREAD

**Issue by fbeleznay** [numworks/epsilon#2274]
#### Describe the bug
PLEASE, can you please prioratise fixing this? The IB exam is coming in May and this issue can be something that can block students finding answers, where examiners expect students to rely on the use of an equation solver for questions similar to the screenshot.

The equation solver sometimes does not give a solution to an equation where the upper limit of the integral is the unknown.

#### Screenshots
![screenshot](https://github.com/user-attachments/assets/71ca263f-0a32-4537-962d-8342091490b5)

#### To Reproduce
1. Go to the 'equations' app
2. Type in the equation
3. Scroll down to 'solve'

The calculator shows that it is working, but it takes way too long, for some equations we could not wait until it gives the solution. We could of course stop the calculation on the physical calculator (the emulator simply froze).

#### Expected behavior
On previous operating systems students could choose the search interval before the calculator start searching for the solution. This is no longer available. Narrowing the interval can speed up finding the solution, but this cannot be done in the new operating system. Students can only change the search interval if the calculator returns something.
Probably in most cases the automatic search interval creation is helpful, but PLEASE, at least give students the option to change it. There are equations (like the one on the screenshot), where it takes a lot of time for the calculator to find the solution. The lack of option to manually set the search interval makes a question like this unsolvable on exams.

#### Environment
 - Epsilon version 23.xxx.
 - The platform(s) on which the problem happens: online simulator, actual device, 
 - Normally I advise students to update their operating system regularly, but until this is fixed, we cannot do this.

**Comment by fbeleznay** [comment]
I would like to come back to this issue, as the IB exams are only 3 months away and this is really a blocking issue for students on exam. They cannot really spend minutes to wait for the calculator to give an answer.
Here is a question from an exam practice workbook.

![Image](https://github.com/user-attachments/assets/e1407311-9316-4160-9c54-58ed2aa7582b)

To answer the last part of the question, students need to solve the equation on the following Numworks screenshot.

![Image](https://github.com/user-attachments/assets/802fc66b-79f4-4aae-9c13-7889137b7b2a)

When I try this on the emulator, the webpage crashes. This alone is an indicator for me that something is not working as it should.
When I try this on my calculator, it takes more than a minute to find an answer. For students on exams this is VERY distracting, they cannot afford to wait until the calculator finishes the calculation.

**Comment by fbeleznay** [comment]
An update: I just updated the operating system on my calculator (23.2.5). When I asked the calculator to solve the equation on the screenshot of my previous post, it started to work. It worked for 10 minutes without giving an answer, I had no patience to wait until it does. I am not sure what algorithm is used, but it is clearly not working as it should. Other graphing calculators give an answer in a reasonable time.

**Comment by adri1numworks** [comment]
Hello, 
My colleague just highlighted this issue, I'm very sorry it slipped through our radar. 

Concerning the second example you gave, on version 22 it still takes around 1 minute for the equation to be solved, after you chose Xmin = 0 and Xmax = 10. If this calculation time is acceptable for you, we could imagine to give the opportunity to the student to set the interval before solving the equation. This would help reducing the calculation time.

**Comment by fbeleznay** [comment]
Yes, this would be something that can help. Please, note though that I asked colleagues to check this on their calculator (without of course mentioning Numworks) and one of them said that her Casio fx-9860GIII (which is a very outdated model) gives the answer in 10 seconds. I have also seen a student getting very quick answer on the TI nspire. I still recommend Numworks for my students, because it is much more intuitive, but nevertheless, this is an indication that there may be a problem with this specific algorithm.
I was thinking that instead of the moving back to version 22, the automatic search can be a start (which is indeed helpful in the majority of the cases), but if you could introduce an escape (for example when student presses the back button) to stop the search at any time, then it would be a good solution. This way students can interrupt the process when they get impatient, the calculator could show the same result screen as now, except with "process interrupted, no solution  found so far" or similar message. Students then could see where the calculator was searching currently and modify the domain.

**Comment by adri1numworks** [comment]
OK I replied to your message on Freshdesk, detailing a potential solution. I let you reach out to me there.

**Comment by adri1numworks** [comment]
Hello all,
You can now update your calculator. We made a hotfix above the version 23 (new update is 23.2.6). 

In Equation, if a calculation takes too much time, you can press Back and you will see a window appear, with a "Calculation interrupted" message displayed. There, you can set the interval of search to reduce the resolution time. I close this ticket in consequence, let me know if there is any problem with this solution.
